### PR TITLE
boards: nucleo_f447ze: Added support for the Nucleo-144 F446ZE board

### PR DESCRIPTION
**Pull request by ArtifixPrime**
This has been done mainly by adding support for more peripherals and
changing pin numbering from the existing port for F446RE board.
Additionally, dts/arm/st/f4/stm32f4.dtsi has been modified by adding
USART3 configuration.
It has been tested to work with the provided Zephyr samples.

Signed-off-by: Peter Nimac <nimac.peter@gmail.com>

### CODE REVIEW

**Comment by bfrog** [comment]
This is missing the flash layout like the RE board did, which means mcumgr, mcuboot, and various other flash related samples will fail to work

File: boards/arm/nucleo_f446ze/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+   :align: center
+   :height: 540px
+   :alt: Nucleo F446ZE Morpho connectors (top right)
+
+For mode details please refer to `STM32 Nucleo-144 board User Manual`_.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+- UART_3_TX : PD8
```
Comment (dbkinder) [inline boards/arm/nucleo_f446ze/doc/index.rst]: If you add this rst-class directive and a blank line before the list, it will display the list in multiple columns instead of a long list that runs down the page:
```
.. rst-class:: rst-columns

- UART_3_TX : PD8
- UART_3_RX : PD9
  etc.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
one recommended appearance tweak otherwise looks fine

File: boards/arm/nucleo_f446ze/doc/index.rst (hunk 1 above)
Comment (ArtifixPrime) [reply boards/arm/nucleo_f446ze/doc/index.rst]: Noted. I've never worked with the Sphinx documentation before, so thanks for the tip.

File: dts/arm/st/f4/stm32f4.dtsi [hunk 2]
```diff
@@ -147,6 +147,15 @@
 			label = "UART_2";
 		};
 
+		usart3: serial@40004800 {
```
Comment (erwango) [inline dts/arm/st/f4/stm32f4.dtsi]: Also, please do it in a dedicated commit

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/nucleo_f446ze/nucleo_f446ze.dts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+ * Copyright (c) 2019 Peter Nimac
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/f4/stm32f446Xe.dtsi>
+#include "arduino_r3_connector.dtsi"
+
+<<<<<<< HEAD
```
Comment (erwango) [inline boards/arm/nucleo_f446ze/nucleo_f446ze.dts]: Please resolve merge conflicts.

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/nucleo_f446ze/nucleo_f446ze.dts (hunk 3 above)
Comment (ArtifixPrime) [reply boards/arm/nucleo_f446ze/nucleo_f446ze.dts]: Could you please help me out a bit?
I've checked all of the files and on my local machine git status returns:
"nothing to commit, working tree clean"

Shippable still fails some checks and prints in the console to resolve merge conflicts.

Sorry, I know this may be a silly request but I've never really worked with git before.

Comment (erwango) [reply boards/arm/nucleo_f446ze/nucleo_f446ze.dts]: > Sorry, I know this may be a silly request but I've never really worked with git before.

@ArtifixPrime, that's ok, there's a first time for everything. I won't be able to help today, but Iwill come back to you this week.

Comment (ArtifixPrime) [reply boards/arm/nucleo_f446ze/nucleo_f446ze.dts]: Alright, thank you so much! 
Sorry for replying this late, I am also busy with my own work, so whenever you'll find the time

**Comment by erwango** [comment]
@ArtifixPrime, I'll try to provide feedback this week. Sorry for the delay.

**Comment by erwango** [comment]
@ArtifixPrime, I had a look, and maybe the easiest way will be the following:
-In your current branch, copy the board directory on a location outside zephyr (the usart3 addition in stm32f4.dtsi is now in zephyr so you don't need it anymore)
-Fetch a latest zephyr version and create a **new** branch on top of the master
-Copy the board directory back in zephyr repo
-Push this new branch 

We'll continue work from there.

**Comment by galak** [comment]
@ArtifixPrime any plans to update?

**Comment by nashif** [comment]
stale, closing. Reopen when there are updates...

**Comment by towen** [comment]
it's a shame this never made it in. I would really like to evaluate zephyr on this board 😢 
I have tried copying the board definition files out of ArtifixPrime's repo but I get errors compiling the device tree:
```
$ west build -p auto -b nucleo_f446ze samples/subsys/usb/cdc_acm_composite
-- west build: generating a build system
Including boilerplate (Zephyr base): /home/tom/Git/stm32_rtos_test/zephyr_test/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/tom/Git/stm32_rtos_test/zephyr_test/zephyr/samples/subsys/usb/cdc_acm_composite
-- Zephyr version: 2.6.99 (/home/tom/Git/stm32_rtos_test/zephyr_test/zephyr), build: zephyr-v2.6.0-533-gc68054c9e519
-- Found Python3: /usr/bin/python3.9 (found suitable exact version "3.9.2") found components: Interpreter 
-- Found west (found suitable version "0.11.0", minimum required is "0.7.1")
-- Board: nucleo_f446ze
-- Cache files will be written to: /home/tom/.cache/zephyr
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Using toolchain: zephyr 0.12.4 (/opt/zephyr-sdk-0.12.4)
-- Found dtc: /opt/zephyr-sdk-0.12.4/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: /home/tom/Git/stm32_rtos_test/zephyr_test/zephyr/boards/arm/nucleo_f446ze/nucleo_f446ze.dts
Lexical error: nucleo_f446ze.dts.pre.tmp:632.21-23 Bad integer literal '09'
Lexical error: nucleo_f446ze.dts.pre.tmp:634.9-11 Bad integer literal '08'
Lexical error: nucleo_f446ze.dts.pre.tmp:635.9-11 Bad integer literal '09'
Lexical error: nucleo_f446ze.dts.pre.tmp:638.21-23 Bad integer literal '09'
Lexical error: nucleo_f446ze.dts.pre.tmp:646.21-23 Bad integer literal '09'
Lexical error: nucleo_f446ze.dts.pre.tmp:647.21-23 Bad integer literal '08'
Error: nucleo_f446ze.dts.pre.tmp:665.22-23 syntax error
FATAL ERROR: Unable to parse input tree
CMake Error at /home/tom/Git/stm32_rtos_test/zephyr_test/zephyr/cmake/dts.cmake:214 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  /home/tom/Git/stm32_rtos_test/zephyr_test/zephyr/cmake/app/boilerplate.cmake:571 (include)
  /home/tom/Git/stm32_rtos_test/zephyr_test/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /home/tom/Git/stm32_rtos_test/zephyr_test/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:35 (include_boilerplate)
  CMakeLists.txt:4 (find_package)


-- Configuring incomplete, errors occurred!
```

**Comment by erwango** [comment]
> it's a shame this never made it in. I would really like to evaluate zephyr on this board 
> I have tried copying the board definition files out of ArtifixPrime's repo but I get errors compiling the device tree:
> 

@towen This PR is pretty old and you will face lot of issues if trying to use on v2.6.0
I rather suggest you try to start from existing nucleo_f446re and adapt from there.
